frontend/src/hooks/Pagination/usePagination.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo, useState } from 'react';
+
+interface paginationProps {
+  totalItemCount: number;
+  perPage: number;
+}
+
+export const usePagination = ({ totalItemCount, perPage }: paginationProps) => {
+  const [page, setPage] = useState(0);
+  const totalPages = useMemo(() => Math.ceil(totalItemCount / perPage), [totalItemCount, perPage]);
```

Comment:
[L3-중요질문]
이렇게 0이하의 perPage가 들어올때는 divzero(0으로 나누기) 처리를 꼭 해줘야 할것 같아요. 지금은 0 이하로 나눌 일이 전혀 없지만, 나중에는 문제 생길수 있으니 막아 놓는것이 좋을것 같네용.

```suggestion
  const totalPages = useMemo(() => {
    if (perPage <= 0) return 1;
    return Math.ceil(totalItemCount / perPage);
  }, [totalItemCount, perPage]);
```